src/pudl/transform/ferceqr.py

```diff
@@ -0,0 +1,187 @@
+"""Transform FERC EQR data."""
+
+import dagster as dg
+import duckdb
+import pandas as pd
+
+from pudl.helpers import (
+    ParquetData,
+    duckdb_relation_from_parquet,
+    persist_table_as_parquet,
+)
+from pudl.metadata.classes import Resource
+from pudl.settings import ferceqr_year_quarters
+
+
+def apply_dtypes_to_duckdb_table(
+    table_name: str,
+    table_data: duckdb.DuckDBPyRelation,
+    **col_expressions: dict[str, duckdb.Expression],
+):
+    """Return a duckdb relation that will cast all columns to appropriate dtypes."""
+    dtypes = Resource.from_id(table_name).to_duckdb_dtypes()
+    return table_data.select(
+        *[
+            expression.cast(dtypes[col]).alias(col)
+            for col, expression in col_expressions.items()
+        ],
+        *[
+            duckdb.ColumnExpression(col).cast(dtype).alias(col)
+            for col, dtype in dtypes.items()
+            if col not in col_expressions
+        ],
+    )
+
+
+def _map_timezones(
+    table: duckdb.DuckDBPyRelation, conn: duckdb.DuckDBPyConnection
+) -> duckdb.DuckDBPyRelation:
+    tz_map = pd.DataFrame(
+        columns=["time_zone", "timezone"],
+        data=[
+            ("EP", "America/New_York"),
+            ("PP", "America/Los_Angeles"),
+            ("ES", "America/New_York"),
+            ("ED", "America/New_York"),
+            ("CP", "America/Chicago"),
+            ("MS", "America/Denver"),
+            ("MD", "America/Denver"),
+            ("CD", "America/Chicago"),
+            ("CS", "America/Chicago"),
+            ("PS", "America/Los_Angeles"),
+            ("PD", "America/Los_Angeles"),
+            ("MP", "America/Denver"),
+            ("EPT", "America/New_York"),
+            ("CPT", "America/Chicago"),
+            ("CST", "America/Chicago"),
+            ("CDT", "America/Chicago"),
+            ("PPT", "America/Los_Angeles"),
+            ("EDT", "America/New_York"),
+            ("AP", "America/New_York"),
+            ("MPT", "America/Denver"),
+            ("N/A", None),
+            ("NA", None),
+        ],
+    )
+    return table.join(conn.from_df(tz_map), condition="time_zone", how="left")
+
+
+def _na_to_null(col_name: str) -> duckdb.CaseExpression:
+    return duckdb.CaseExpression(
+        condition=duckdb.ColumnExpression(col_name) in ["N/A", "NA"],
```

Comment:
How does this handle getting "n/a" or "na"? The old script forced everything to be upper case explicitly before doing the check. There were definitely non-uniform cases in the inputs.

It's not clear how this translates the null-representing strings to actual nulls.

Reply:
Personally I think the `duckdb.SQLExpression()` syntax for these guys is much simpler and more readable.

Reply:
Switched to the `duckdb.SQLExpression()`

Reply:
Would you also be willing to switch the other Y/N to boolean case statements to use`duckdb.SQLExpression()`?

I think we're also going to have some more complex case statements (next PR) to handle the mapping of non-standard ENUM values to their canonical values.